messages/java/src/main/java/io/cucumber/messages/Messages.java

```diff
@@ -0,0 +1,4018 @@
+        public Timestamp getTimestamp() {
+            return timestamp;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            TestCaseStarted that = (TestCaseStarted) o;
+            return 
+                attempt.equals(that.attempt) &&             
+                id.equals(that.id) &&             
+                testCaseId.equals(that.testCaseId) &&             
+                timestamp.equals(that.timestamp);            
+        }
+
+        @Override
+        public int hashCode() {
+            return java.util.Objects.hash(
+                attempt,
+                id,
+                testCaseId,
+                timestamp
+            );
+        }
+
+        @Override
+        public String toString() {
+            return "TestCaseStarted{" +
+                "attempt=" + attempt +
+                ", id=" + id +
+                ", testCaseId=" + testCaseId +
+                ", timestamp=" + timestamp +
+                '}';
+        }
+    }
+
+    public static class TestRunFinished {
+        private final String message;
+        private final Boolean success;
+        private final Timestamp timestamp;
+
+        public TestRunFinished(
+            String message,
+            Boolean success,
+            Timestamp timestamp
+        ) {
+            this.message = message;
+            this.success = java.util.Objects.requireNonNull(success, "TestRunFinished.success cannot be null");
+            this.timestamp = java.util.Objects.requireNonNull(timestamp, "TestRunFinished.timestamp cannot be null");
+        }
+
+        public java.util.Optional<String> getMessage() {
+            return java.util.Optional.ofNullable(message);
+        }
+
+        public Boolean getSuccess() {
+            return success;
+        }
+
+        public Timestamp getTimestamp() {
+            return timestamp;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            TestRunFinished that = (TestRunFinished) o;
+            return 
+                java.util.Objects.equals(message, that.message) &&             
+                success.equals(that.success) &&             
+                timestamp.equals(that.timestamp);            
+        }
+
+        @Override
+        public int hashCode() {
+            return java.util.Objects.hash(
+                message,
+                success,
+                timestamp
+            );
+        }
+
+        @Override
+        public String toString() {
+            return "TestRunFinished{" +
+                "message=" + message +
+                ", success=" + success +
+                ", timestamp=" + timestamp +
+                '}';
+        }
+    }
+
+    public static class TestRunStarted {
```

Comment:
Consider using `public static final` to reduce the api surface.

Reply:
Fixed!